src/coreclr/jit/importer.cpp

```diff
@@ -4387,6 +4387,42 @@ GenTree* Compiler::impImportStaticFieldAccess(CORINFO_RESOLVED_TOKEN* pResolvedT
     FieldSeq::FieldKind fieldKind =
         isSharedStatic ? FieldSeq::FieldKind::SharedStatic : FieldSeq::FieldKind::SimpleStatic;
 
+    // In case of boxed statics if we have a direct pointer to a boxed static's content
+    // we can import it as a GT_OBJ(CNS_INT)/GT_IND(CNS_INT)
+    if (pFieldInfo->boxedStaticContent != nullptr)
```

Comment:
Why duplicate code for this? Ideally, `pFieldInfo` would have a `fieldAddress` field for all known-address statics like these ones, which we'd just use below instead of calling `getFieldAddress`, but even without that, we can just use `boxedStaticContent` as that known address (if non-null).